### DISCUSSION TITLE
Error: [Vuex Electron] Not connected to renderer.

### DISCUSSION THREAD

**Discussion by JonasDes** [BetaHuhn/vuex-electron-store#92] [Q&A] [answered]
I'am using NuxtJS.

### Main
```typescript
 this.vuex = PersistedState.getStoreFromRenderer()
 console.log(this.vuex.getState());
```

### Render (As NuxtJS Plugin)

```typescript
import Vue from 'vue'
import Vuex from 'vuex'

import PersistedState from 'vuex-electron-store'

Vue.use(Vuex)

export default new Vuex.Store({

  state: {
    username: 'Jonas'
  },
  plugins: [
    PersistedState.create()
  ],
})
```

### or as NuxtJS Store
```typescript
import PersistedState from "vuex-electron-store"

export const state = () => ({
  username: 'Jonas'
})

export const plugins = [
  PersistedState.create({ dev: true, ipc: true, paths: ['username'] })

]
```

**Comment by BetaHuhn** [comment]
Does it work in the last example ("as NuxtJS Store")?

The first one ("As NuxtJS Plugin") is missing the `ipc` option.

**Reply by JonasDes** [reply]
Nope, won't work either.
I'll try the  NuxtJS Plugin-Version with IPC and come back

**Reply by JonasDes** [reply]
Nope. Sadly nothing changed.

**Reply by BetaHuhn** [reply]
One thing I noticed is that you are not using `await` when calling `.getState()`. It returns a promise.

But that shouldn't cause that error.

**Reply by JonasDes** [reply]
I've used `await` before, but it don't cause the error

**Reply by BetaHuhn** [reply]
One thing that could be the cause is a race condition. If you call `.getState()` before the IPC connection between the main and renderer process is established, it will cause that error.

Could you try to call `.getState()` after a delay? Like this:

```js
 this.vuex = PersistedState.getStoreFromRenderer()

setTimeout(() =>  console.log(this.vuex.getState()), 1000)
```

**Reply by JonasDes** [reply]
Good idea! Was my first too, but nothing changed.
Can i check that the IPC is connect anywhere?

**Reply by BetaHuhn** [reply]
Okay, then we can at least rule that one out. 

You can try https://www.electronjs.org/devtron

**Reply by BetaHuhn** [reply]
I have one project where I'm using `vuex-electron-store` with Nuxt, I'll see how I'm doing it there.

**Reply by JonasDes** [reply]
Devtron is sadly no longer maintained.
I'm using Electron 13

**Reply by BetaHuhn** [reply]
Ah didn't know that.

Can you see if one of the other methods work? Like `.commit()`

**Reply by BetaHuhn** [reply]
Okay I can reproduce the error now. Calling `.getState` after `PersistedState.getStoreFromRenderer()` will throw that error.

The other methods work for me.

**Reply by BetaHuhn** [reply]
I'll work on a fix and get back to you.

**Reply by BetaHuhn** [reply]
This works:

```js
setTimeout(async () => {
	const data = await store.getState()
	console.log(data)
}, 5000)
```

But only with a timeout of > 4 sec.

**Reply by JonasDes** [reply]
Öhm. Not for me... How did u handle the Nuxt shit?

**Reply by BetaHuhn** [reply]
Found the problem. Releasing a fix now.

Shouldn't be a problem on your side.

**Reply by BetaHuhn** [reply]
Okay, it should be fixed in [v1.4.13](https://github.com/BetaHuhn/vuex-electron-store/releases/tag/v1.4.13).

`PersistedState.getStoreFromRenderer()` now waits for the renderer to connect and the returns a promise containing the store:

```js
const store = await PersistedState.getStoreFromRenderer()

const state = await store.getState()
console.log(state)
```

Let me know if it works @JonasDes!

**Comment by BetaHuhn** [comment, marked as answer]
Fixed in [v1.4.13](https://github.com/BetaHuhn/vuex-electron-store/releases/tag/v1.4.13).

`PersistedState.getStoreFromRenderer()` now waits for the renderer to connect and the returns a promise containing the store:

```js
const store = await PersistedState.getStoreFromRenderer()

const state = await store.getState()
console.log(state)
```
